src/operator/mshadow_op.h

```diff
@@ -1087,10 +593,10 @@ struct smooth_l1_gradient {
   template<typename DType>
   MSHADOW_XINLINE static DType Map(DType a, DType b) {
     b *= b;
-    if (a > DType(1.0f) / b) {
-      return DType(1.0f);
-    } else if (a < DType(-1.0f) / b) {
-      return DType(-1.0f);
+    if (a > 1.0f / b) {
+      return 1.0f;
```

Comment:
`return DType(1)` consistent with the other condition branch?

Reply:
Yes you are right, this code is quite weird. But since my previous changes gave somebody trouble, I'd like to roll them back here.

Reply:
I see. Thanks.